broadcast/internal/outter_errors.go

```diff
@@ -0,0 +1,44 @@
+package outter_errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+func New(msg string, inner error) error {
+	return &outterErr{
+		msg:   msg,
+		inner: inner,
+	}
+}
+
+type outterErr struct {
```

Comment:
I don't really get, what we want to achieve with this component? How is it better then just `fmt.Errorf`?

Reply:
Then I would suggest:
1. We should have tests for what we expect from using it, so it would be visible
2. I'm not sure if having the indentation for each level won't make it hard to read at some point.
3. I would rather use a word `Cause` or `Caused by` instead of `inner error`
4. Maybe simple function like that would be enough:
```
func WithCause(err error, cause error) error {
	return errors.Join(err, fmt.Errorf("\tcaused by: %w\t", cause))
}
```
And using this like:
```
err := WithCause(error.New(...), err)
log.Error().Msgf(..., err)
```
could produce the message like:
```
2024-02-02 08:44:04.295 ERR SubmitTransaction: submitting failed: submitting failed
	caused by: rawTxRequest: bt.NewTxFromString failed
	caused by: script(27): got 21 bytes: unexpected EOF		 application=pulse-default ecs.version=1.6.0
```

Reply:
good enough for me

Reply:
for me too 👍